Aikido.Zen.Core/Helpers/IPHelper.cs

```diff
@@ -7,6 +7,58 @@ namespace Aikido.Zen.Core.Helpers
 {
     public class IPHelper
     {
+        private static readonly IPRange _privateIpRanges = InitializePrivateRanges();
+
+        private static IPRange InitializePrivateRanges()
+        {
+            var ranges = new IPRange();
+
+            // IPv4 private ranges - updated to include all reserved ranges
+            // taken from https://github.com/frenchbread/private-ip/blob/master/src/index.ts
+            var ipv4Ranges = new[]
+            {
+                "0.0.0.0/8",        // "This" network
+                "10.0.0.0/8",       // Private network
+                "100.64.0.0/10",    // Shared Address Space
+                "127.0.0.0/8",      // Loopback
+                "169.254.0.0/16",   // Link-local
+                "172.16.0.0/12",    // Private network
+                "192.0.0.0/24",     // IETF Protocol Assignments
+                "192.0.2.0/24",     // TEST-NET-1
+                "192.31.196.0/24",  // Additional reserved range
+                "192.52.193.0/24",  // Additional reserved range
+                "192.88.99.0/24",   // 6to4 Relay Anycast
+                "192.168.0.0/16",   // Private network
+                "192.175.48.0/24",  // Additional reserved range
+                "198.18.0.0/15",    // Network Interconnect Device Benchmark Testing
+                "198.51.100.0/24",  // TEST-NET-2
+                "203.0.113.0/24",   // TEST-NET-3
+                "224.0.0.0/4",      // Multicast
+                "240.0.0.0/4",      // Reserved for future use
+                "255.255.255.255/32" // Limited broadcast
+            };
+
+            // IPv6 private ranges - updated to be more specific
+            var ipv6Ranges = new[]
+            {
+                "::/128",           // Unspecified address
+                "::1/128",          // Loopback address
+                "fc00::/7",         // Unique local address (ULA)
+                "fe80::/10",        // Link-local address
+                // "ff00::/8",         // Multicast, leaving this out for now, since it can include private and public ranges --> https://datatracker.ietf.org/doc/html/rfc4291#section-2.7
+                "::ffff:0:0/96",    // IPv4-mapped addresses
```

Comment:
Could include public and private IP addresses. Searching a good fix, also an issue in all other agents.